src/python/pants/option/option_types.py

```diff
@@ -0,0 +1,481 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from enum import Enum
+
+# NB: Avoid using "Optional" and use "| None" instead, as "Option" and "Optional" look similar.
```

Comment:
I don't think this comment is necessary. We already try to use `|` throughout the codebase. Anywhere we use `Optional` is legacy.